BraveRewardsTests/LedgerDatabaseTests.mm

```diff
@@ -144,6 +147,638 @@ - (void)testNonZeroExcludedCount
+                                    excluded:(BATPublisherExclude)excluded
+{
+  
+  const auto info = [[BATPublisherInfo alloc] init];
+  info.id = publisherId;
+  info.reconcileStamp = stamp;
+  info.percent = percent;
+  info.duration = duration;
+  info.visits = visits;
+  info.excluded = excluded;
+  
+  [self backgroundSaveAndWaitForExpectation:^{
+    [BATLedgerDatabase insertOrUpdatePublisherInfo:info];
+  }];
+  
+  if(createActivityInfo){
+    [self backgroundSaveAndWaitForExpectation:^{
+      [BATLedgerDatabase insertOrUpdateActivityInfoFromPublisher:info];
+    }];
+  }
+  
+  return info;
+}
+
+- (void) testInsertOrUpdatePublisherInfoEmptyId
+{
+  const auto info = [[BATPublisherInfo alloc] init];
+  info.id = @"";
+  
+  [self waitForFailedSaveAttempt:^{
+    [BATLedgerDatabase insertOrUpdatePublisherInfo:info];
+  }];
+  
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"PublisherInfo"]);
+}
+
+- (BOOL)countMustBeEqualTo:(int)count forEntityName:(NSString *)name
+{
+  const auto fetchRequest = [[NSFetchRequest alloc] initWithEntityName:name];
+  NSError *error;
+  const auto recordsCount = [[DataController viewContext] countForFetchRequest:fetchRequest error:&error];
+  
+  return count == recordsCount;
+}
+
+#pragma mark - Contribution Info
+
+- (void)testInsertContributionInfo
+{
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"ContributionInfo"]);
+  
+  [self makeOneTimeTip:@"333" month:10 year:2020];
+  
+  XCTAssert([self countMustBeEqualTo:1 forEntityName:@"ContributionInfo"]);
+}
+
+- (void)testOneTimeTipsPublishersForMonth
+{
+  BATActivityMonth month = BATActivityMonthJanuary;
+  const auto year = 2019;
+  const auto publisherId = @"333";
+  
+  [self makeOneTimeTip:publisherId month:month year:year];
+  
+  // Tip with different month
+  [self makeOneTimeTip:publisherId month:BATActivityMonthFebruary year:year];
+  
+  // Tip with different year
+  [self makeOneTimeTip:publisherId month:month year:2020];
+  
+  // Tip with different category
+  [self makeOneTimeTip:publisherId month:month year:year
+                 probi:@"200000000000000" category:BATRewardsCategoryAutoContribute];
+  
+  const auto tips = [BATLedgerDatabase oneTimeTipsPublishersForMonth:month year:year];
+  XCTAssert([self countMustBeEqualTo:4 forEntityName:@"ContributionInfo"]);
+  XCTAssertEqual(tips.count, 1);
+  
+  // Add two more valid tips
+  
+  // Same publisher
+  [self makeOneTimeTip:publisherId month:BATActivityMonthJanuary year:year];
+  
+  // Different publisher
+  [self makeOneTimeTip:@"444" month:BATActivityMonthJanuary year:year];
+  
+  const auto newerTips = [BATLedgerDatabase oneTimeTipsPublishersForMonth:month year:year];
+  XCTAssert([self countMustBeEqualTo:6 forEntityName:@"ContributionInfo"]);
+  XCTAssertEqual(newerTips.count, 3);
+}
+
+- (void)makeOneTimeTip:(NSString *)publisherId month:(const int)month year:(const int)year
+{
+  const auto probi = @"10000000000000000";
```

Comment:
Since this method is a duplicate of the one below minus some arguments, it should call the other with said default arguments.

Example:

```objc
- (void)makeOneTimeTip:(NSString *)publisherId month:(const int)month year:(const int)year
{
  [self makeOneTimeTip:publisherId month:month year:year probi:@"10000000000000000" category:BATRewardsCategoryOneTimeTip];
}
```